api/scripts/update-users-with-shouldValidateTermsOfService-to-true.js

```diff
@@ -0,0 +1,52 @@
+const _ = require('lodash');
+const Bookshelf = require('../lib/infrastructure/bookshelf');
+
+const BookshelfUser = require('../lib/infrastructure/data/user');
+
+async function updateShouldValidateTermsOfServiceWhenUsersHaveNotSamlIdAndNotAssociatedWithSchoolingRegistration() {
+  const subquery = Bookshelf.knex
+    .select('users.id')
+    .from('users')
+    .leftJoin('schooling-registrations', 'users.id', 'schooling-registrations.userId')
+    .where({
+      'users.samlId': null,
+      'schooling-registrations.userId': null
+    });
+
+  const foundUsers = await BookshelfUser
+    .query((qb) => {
+      qb.update({ shouldValidateTermsOfService: true }, ['id'])
+        .whereIn('id', subquery);
+    })
+    .fetchAll();
+
+  return _.map(foundUsers.models, 'id');
```

Comment:
Un `fetchAll` de Bookshelf avec un `update` ? C'est un peu exagéré comme détournement je trouve.

Si on voulait vraiment la liste des `id` :

```suggestion
  return Bookshelf.knex
    .update({ shouldValidateTermsOfService: true })
    .whereIn('id', subquery)
    .returning('id');
```

Mais bon, remonter plusieurs centaines de milliers d'`id` si c'est juste pour savoir combien on en a mis à jour, c'est… dommage.

Pour avoir juste le nombre de lignes, il suffit d'enlever le `returning` :

```suggestion
  return Bookshelf.knex
    .update({ shouldValidateTermsOfService: true })
    .whereIn('id', subquery);
```

Reply:
L'id est aussi remonté pour vérifier dans le test (update-users-with-shouldValidateTermsOfService-to-true_test.js) que l'enregistrement a bien été mis à jour. 

On pourrait s'en passer si la vérification était faite en BDD, mais je ne sais pas si c'est une bonne idées. Les autres tests fonctionnent de la même manière (send-invitations-to-sco-organizations_test.js)

Reply:
Dans tous les cas, j'ai adopté ta suggestion pour filtrer l'id à la source

Reply:
Je suggère de modifier le test pour faire la vérification en base de données. Il serait ainsi plus symétrique :
 * _Given_ des lignes en base de données (`databaseBuilder`)
 * _When_ …
 * _Then_ les lignes ont été modifiées / pas modifiées.

Là on a le test qui exploite un détail d'implémentation de la fonction pour faire une vérification qui n'est même pas très convaincante — la fonction en question pourrait renvoyer (par erreur) le résultat du `select` sans avoir fait l'`update` et le test passerait 🤦